controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -335,9 +336,20 @@ func recordActionFailure(info *reconcileInfo, errorType metal3v1alpha1.ErrorType
 	return actionFailed{dirty: true, ErrorType: errorType, errorCount: info.host.Status.ErrorCount}
 }
 
-func recordActionDelayed(info *reconcileInfo) actionResult {
+func recordActionDelayed(info *reconcileInfo, nextState metal3v1alpha1.ProvisioningState) actionResult {
+	var counter prometheus.Counter
+
+	switch nextState {
+	case metal3v1alpha1.StateInspecting, metal3v1alpha1.StateProvisioning:
+		counter = delayedProvisioningHostCounters.With(hostMetricLabels(info.request))
+	case metal3v1alpha1.StateDeprovisioning, metal3v1alpha1.StateDeleting:
+		counter = delayedDeprovisioningHostCounters.With(hostMetricLabels(info.request))
+	case metal3v1alpha1.StateReady:
+		counter = delayedProvisioningHostCounters.With(hostMetricLabels(info.request))
+	default:
+		return actionDelayed{}
```

Comment:
This doesn't sound right to me. The operationalStatus must be set regardless of the provisioning state. I'd suggest just to keep a trace here

Reply:
I am not following, you mean that for `default` case I still need to set `counter` ?
To what value then? How to choose is it a provisioning or deprovisioning counter?
Also I need to define counter because it gets incremented in few lines below, without setting it, nil pointer exception occurs.

Reply:
@andfasano Your feedback is highly appreciated here, thanks.

Reply:
The `default` case returns immediately with `actionDelayed{}` skipping the operationalStatus. On a second though, I think a more conservative approach (in respect to the previous implementation) here could be preferrable